zerver/lib/bugdown/__init__.py

```diff
@@ -77,6 +77,16 @@
 class BugdownRenderingException(Exception):
     pass
 
+def rewrite_if_relative_link(link: str) -> str:
+    """ If the link points to a local destination we can just switch to that
+    instead of opening a new tab.
+    """
+    if db_data:
+        relative_link = re.match("^{}\/(#.+)$".format(re.escape(db_data['realm_uri'])), link)
```

Comment:
I don't like the idea of interpolating a uri into a regex, even though you do take the precaution of `re.escape` here.  I think we already know that the uri is a valid link, so why not use `startswith` to test if it's relative, and then just slice off the beginning?

Reply:
P.S. I guess you inherited some of this code, but we should re-think it now that I'm looking at it.  Also, we should try to extract functions in separate commits when possible.